crates/firehose-protos/src/solana/sol_block.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::Block;
+use firehose_rs::{Response, SingleBlockResponse};
+use prost::Message;
+use prost_wkt_types::Any;
+
+use crate::error::ProtosError;
+
+fn decode_block<M>(response: M) -> Result<Block, ProtosError>
+where
+    M: MessageWithBlock,
+{
+    let any = response
+        .block()
+        .ok_or(ProtosError::BlockMissingInResponse)?;
+    let block = Block::decode(any.value.as_ref())?;
+    Ok(block)
+}
+trait MessageWithBlock {
```

Comment:
```suggestion

trait MessageWithBlock {
```

Really no big deal but I would leave a line there